lib/pure/strutils.nim

```diff
@@ -339,6 +339,25 @@ proc strip*(s: string, leading = true, trailing = true,
     while last >= 0 and s[last] in chars: dec(last)
   result = substr(s, first, last)
 
+proc countWhile*(str: string, pred : proc(a:char):bool, prefix = true): int =
```

Comment:
Nitpick: Watch the spacing around your colons.

Reply:
``prefix`` should instead be ``countdown=false``.

Reply:
And I think ``s`` instead of ``str`` is more common in the stdlib.

Reply:
And I think the signature should be this, there is little you can do in this callback that receives a single ``char``...

```nim

proc countWhile*(s: string; x: set[char]; countdown = false): int
```

Or maybe 

```nim

template countWhileIt*(s: string; e: untyped; countdown = false): int
```